src/gui/speedlimitdialog.cpp

```diff
@@ -37,10 +38,21 @@ SpeedLimitDialog::SpeedLimitDialog(QWidget *parent)
 {
     m_ui->setupUi(this);
 
-    // Connect to slots
-    connect(m_ui->bandwidthSlider, &QSlider::valueChanged, this, &SpeedLimitDialog::updateSpinValue);
-    connect(m_ui->spinBandwidth, qOverload<int>(&QSpinBox::valueChanged)
-            , this, &SpeedLimitDialog::updateSliderValue);
+    BitTorrent::Session *const session = BitTorrent::Session::instance();
+    const bool isAltGlobalLimits = session->isAltGlobalSpeedLimitEnabled();
```

Comment:
ideally this class shouldn't care about alt limits or not, isn't?

Reply:
The dialog should only care about the normal global limits? As for Session, I'd need to pass 2 more vars in the struct, that's what i did at first but then i looked at updownratiodialog and saw it uses Session as well and thought: oh great i can just do that then.

Reply:
>The dialog should only care about the normal global limits? 

No, not normal limits nor alternate limits, just limits. Only the client code using this dialog should know which limit (normal or alternate) it is operating on.

>As for Session, I'd need to pass 2 more vars in the struct, 

Seems so, you need to add 2 max limit variables for upload and download.

>updownratiodialog and saw it uses Session as well and thought: oh great i can just do that then.

Well that is not a preferable way to do it IMO.

Reply:
and done.